yaserde_derive/src/primitives.rs

```diff
@@ -0,0 +1,123 @@
+// Adds YaSerialize and YaDeserialize implementations for types that support FromStr and Display traits.
+// Code originally from `xsd-parser-rs`
+
+use proc_macro::TokenStream;
+use proc_macro2::TokenStream as TokenStream2;
+use quote::quote;
+use syn::{parse_macro_input, DeriveInput};
+
+pub fn primitive_yaserde(input: TokenStream) -> TokenStream {
+    let ast = parse_macro_input!(input as DeriveInput);
+
+    let struct_name = &ast.ident;
+    let struct_name_literal = &ast.ident.to_string();
+
+    let serde = quote! {
+        impl ::yaserde::YaSerialize for #struct_name {
+            fn serialize<W: ::std::io::Write>(
+                &self,
+                writer: &mut ::yaserde::ser::Serializer<W>,
+            ) -> ::std::result::Result<(), ::std::string::String> {
+              ::yaserde::primitives::serialize_primitives(
+                    self,
+                    #struct_name_literal,
+                    writer, |s| s.to_string(),
+                )
+            }
+
+            fn serialize_attributes(
+                &self,
+                attributes: ::std::vec::Vec<::yaserde::__xml::attribute::OwnedAttribute>,
+                namespace: ::yaserde::__xml::namespace::Namespace,
+            ) -> ::std::result::Result<
+                (
+                    ::std::vec::Vec<::yaserde::__xml::attribute::OwnedAttribute>,
+                    ::yaserde::__xml::namespace::Namespace,
+                ),
+                ::std::string::String,
+            > {
+                Ok((attributes, namespace))
+            }
+        }
+
+        impl ::yaserde::YaDeserialize for #struct_name {
+            fn deserialize<R: ::std::io::Read>(
+                reader: &mut ::yaserde::de::Deserializer<R>,
+            ) -> ::std::result::Result<Self, ::std::string::String> {
+                ::yaserde::primitives::deserialize_primitives(
+                    reader,
+                    |s| #struct_name::from_str(s).map_err(|e| e.to_string()),
+                )
+            }
+        }
+    };
+
+    serde.into()
+}
+
+pub fn hexbinary_serde(input: TokenStream) -> TokenStream {
+    let first = input.clone();
+    let DeriveInput { ident, .. } = parse_macro_input!(first);
+    // Calculate number digits to determine whether leading zero should be added
+    quote! {
+      impl std::fmt::Display for #ident {
+        fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+          let digits = (self.bits() as f64).log(16.0).ceil() as usize;
+          if digits % 2 == 0 {
+            write!(f, "{:X}", self.0)
+          } else {
+            write!(f,"0{:X}", self.0)
+          }
```

Comment:
usage of std format address the case of formatting with a padding of 2 chars using: `{:02x}`

So you may replace these 5 lines with:
```rust
write!(f, "{:02X}", self.0)
```